packages/@aws-cdk/aws-kinesisanalyticsv2/lib/flink-application.ts

```diff
@@ -0,0 +1,328 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as logs from '@aws-cdk/aws-logs';
+import * as core from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { ApplicationCode } from './application-code';
+import { CfnApplication, CfnApplicationCloudWatchLoggingOption } from './kinesisanalyticsv2.generated';
+import { flinkApplicationConfiguration, FlinkLogLevel, FlinkMetricsLevel } from './private/flink-application-configuration';
+import { validateFlinkApplicationProps } from './private/validation';
+import { PropertyGroup } from './property-group';
+
+export interface IFlinkApplication extends core.IResource, iam.IGrantable {
+  /**
+   * The application ARN.
+   *
+   * @attribute
+   */
+  readonly applicationArn: string;
+
+  /**
+   * The name of the Flink application.
+   *
+   * @attribute
+   */
+  readonly applicationName: string;
+
+  /**
+   * The application IAM role.
+   */
+  readonly role?: iam.IRole;
+
+  /**
+   * Convenience method for adding a policy statement to the application role.
+   */
+  addToPrincipalPolicy(policyStatement: iam.PolicyStatement): boolean;
+}
+
+/**
+ * Available Flink runtimes for Kinesis Analytics.
+ *
+ * @example
+ * // Creating a new runtime that isn't in CDK yet.
+ * const runtime = new FlinkRuntime(FLINK-9_99);
+ */
+export class FlinkRuntime {
+  public static readonly FLINK_1_6 = new FlinkRuntime('FLINK-1_6');
+  public static readonly FLINK_1_8 = new FlinkRuntime('FLINK-1_8');
+  public static readonly FLINK_1_11 = new FlinkRuntime('FLINK-1_11');
+  public constructor(public readonly value: string) {}
+}
+
+/**
+ * Implements the functionality shared between CDK created and imported
+ * IFlinkApplications.
+ */
+abstract class FlinkApplicationBase extends core.Resource implements IFlinkApplication {
+  public abstract readonly applicationArn: string;
+  public abstract readonly applicationName: string;
+  public abstract readonly role?: iam.IRole;
+
+  // Implement iam.IGrantable interface
+  public abstract readonly grantPrincipal: iam.IPrincipal;
+
+  /** Implement the convenience {@link IFlinkApplication.addToPrincipalPolicy} method. */
+  public addToPrincipalPolicy(policyStatement: iam.PolicyStatement): boolean {
+    if (this.role) {
+      this.role.addToPrincipalPolicy(policyStatement);
+      return true;
+    }
+
+    return false;
+  }
+}
+
+/**
+ * Props for creating a FlinkApplication construct.
+ */
+export interface FlinkApplicationProps {
+  /**
+   * A name for your FlinkApplication that is unique to an AWS account.
+   *
+   * @default - CloudFormation-generated name
+   */
+  readonly applicationName?: string;
+
+  /**
+   * The Flink version to use for this application.
+   */
+  readonly runtime: FlinkRuntime;
```

Comment:
I was concerned that providing a default here would result in breaking changes or retaining an old default for too long in CDK.

If I write this code and get FLINK_1_11

```ts
const new ka.FlinkApplication(this, 'App', { code });
```

then when I update CDK I would be sad if the same code redeployed my application with FlINK_1_12 because I'm not very confident that wouldn't break my app. If I explicitly pick a version (which I probably used to build the JAR and write tests with) then I can be more confident that CDK upgrades won't break my app.